### PR TITLE
Adds a lot of missing resource templates

### PR DESCRIPTION
**Pull request by TWinsnes**
Added templates:
* CloudFront distributions
* Autoscaling groups
* global and regional WAFv2
* SNS topics
* RDS cluster, instance option groups, snapshots and subnet groups
* KMS keys
* EKS clusters and workernode groups
* Backup vault
* Kinesis delivery stream

### CODE REVIEW

File: link2aws.js [hunk 1]
```diff
@@ -53,6 +54,8 @@ class ARN {
             throw Error("Bad number of tokens");
         }
 
+        console.log(this);
```
Comment (fxkr) [inline link2aws.js]: Can you remove the log statement? As a library, we should not log.

File: link2aws.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ class ARN {
         }
 
         // arn:partition:service:region:account-id:...
+        this.arn = text;
```
Comment (fxkr) [inline link2aws.js:15]: Can you update the tests in `test/test.js` that check all the fields to check the new `.arn` field as well?

File: link2aws.js [hunk 3]
```diff
@@ -187,14 +190,17 @@ class ARN {
                 "workgroup": null,
             },
             "autoscaling": { // Amazon EC2 Auto Scaling
-                "autoScalingGroup": null,
+                "autoScalingGroup": () => {
+                    const groupName = this.resource.split('/')[1];
+                    return `https://${this.region}.console.aws.amazon.com/ec2/home?region=${this.region}#AutoScalingGroupDetails:id=${groupName};view=details`
```
Comment (fxkr) [inline link2aws.js]: In all places where `console.aws.amazon.com` is hardcoded, can you use `${this.console}` instead? The console domain is different for AWS GovCloud and AWS China.

**Review by fxkr — CHANGES_REQUESTED** [review]
Thank you very much for all of these! I really appreciate it!

I do have a few review comments. Could you update your commits to address these? Thank you!

**Comment by TWinsnes** [comment]
Thanks for reviewing so quickly, good call outs, I was obviously a bit sloppy there. 
I've fixed up the issues you pointed out

**Review by fxkr — APPROVED** [review]

**Comment by fxkr** [comment]
Thank you too for getting back to me so quickly! And thanks again for the contributions.

Released as version 1.0.18.

**Pull request merged**